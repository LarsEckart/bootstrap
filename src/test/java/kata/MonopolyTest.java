package kata;

import com.github.larseckart.tcr.TestCommitRevertExtension;
import org.approvaltests.Approvals;
import org.approvaltests.StoryBoard;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;

import static org.assertj.core.api.Assertions.assertThat;

@ExtendWith(TestCommitRevertExtension.class)
class MonopolyTest {

    @Test
    void testBasicPlay() {
        StoryBoard story = new StoryBoard();
        Monopoly monopoly = new Monopoly(3);
        story.addFrame(monopoly);
        story.addFrame(monopoly.move(5).buyProperty().endTurn());
        story.addFrame(monopoly.move(3).buyProperty().endTurn());
        story.addFrame(monopoly.move(3).endTurn());
        Approvals.verify(story);
    }

    @Test
    void testIncomeTaxAndChanceAndJail() {
        StoryBoard story = new StoryBoard();
        Monopoly monopoly = new Monopoly(3);
        story.addFrame(monopoly);
        story.addFrame(monopoly.move(10).endTurn());
        story.addFrame(monopoly.move(4).endTurn());
        story.addFrame(monopoly.move(7).endTurn());
        Approvals.verify(story);
    }

    @Test
    void testPlayerDetails() {
        Player player = new Player("Player 1");

        assertThat(player.details()).isEqualTo("Player 1 ($1500) ");
    }

    @Test
    void testPlayerLocation() {
        Player player = new Player("Player 1");

        assertThat(player.location()).isEqualTo(0);
    }

}
