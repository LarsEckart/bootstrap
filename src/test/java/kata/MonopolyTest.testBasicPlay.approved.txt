Initial:
|          GO          | [Player 1, Player 2, Player 3] |
| Mediterranean Avenue |               []               |
|   Community Chest    |               []               |
|    Baltic Avenue     |               []               |
|      Income Tax      |               []               |
|   Reading Railroad   |               []               |
Player 1 ($1500)
Player 2 ($1500)
Player 3 ($1500)
